### PR TITLE
feat: add Letter combinations of phone book problem (#17)

### PR DESCRIPTION
**Pull request by mindaugl**
#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: leetcode/src/17.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                    ind = k * l_tot * strlen(cp) + l + l_tot * j;
+                    combs[ind][i] = cp[j];
+                }
+            }
+        }
+        k_tot *= strlen(cp);
+    }
+
+    return combs;
+}
```
Comment (Panquesito7) [inline leetcode/src/17.c:71]: Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: leetcode/src/17.c (hunk 1 above)
Comment (mindaugl) [reply leetcode/src/17.c]: Just pushed a new commit with newline

**Review by Panquesito7 — DISMISSED** [review]
Looks great. Thanks! 🚀

**Comment by alexpantyukhin** [comment]
Could you add please small description of the main algorithm idea, runtime complexity and space complexity?

**Comment by mindaugl** [comment]
> Could you add please small description of the main algorithm idea, runtime complexity and space complexity?

Just added brief description of the algorithm, thanks

**Review by Panquesito7 — APPROVED** [review]

**Review by alexpantyukhin — APPROVED** [review]

**Pull request merged**